### DISCUSSION TITLE
Juvix CLI Needs a REPL

### DISCUSSION THREAD

**Discussion by alhassy** [anoma/juvix#1511] [Ideas]
**Is your feature request related to a problem? Please describe.**

When diving into a language, it can be nice to have a REPL to quickly experiment with APIs or language features.
- Something like Haskell's GHCI  (since Juvix is clearly inspired by it) or NodeJS (since Juvix interoperates with it) would be nice.

If a REPL already exists, I couldn't figure it out from the CLI's docs nor the repo's README, and so improved exposure would be appreciated.

**Describe the solution you'd like**

For a CLI option to be added to provide an interactive session where definitions can be declared and expressions are evaluated.

**Describe alternatives you've considered**

A quick and dirty alternative is to *build an interactive session using a temporary file*:
1. Take the currently selected text in your editor, write that to a temporary file, then compile that file. 
2. Then subsequent selections for evaluation are appended to the aforementioned temporary file, that is then compiled and the result is echoed to the user in their editor's UI.
3. If the run succeeds then that candidate text is saved into the temporary file, otherwise it is not.

The benefit here is that this is doable without altering the Juvix CLI, but it is very clunky, editor-specific, and so far from ideal.

**Additional context**
This issue caused #1484

**Comment by janmasrovira** [comment]
Hey @alhassy, currently juvix does not have a repl but we would love to have one. However, there is a lot of work to do in the backend of the compiler and we will be busy with that for a long while. After that, we may consider the option of adding a repl, but we cannot give any approximation of when (or if) it will happen.

**Comment by jonaprieto** [comment]
Since Juvix v0.2.9, we can play with two REPL (one for the user, one for devs)
- ``` 
  juvix repl
  ```
- ```
  juvix dev core repl
  ```
